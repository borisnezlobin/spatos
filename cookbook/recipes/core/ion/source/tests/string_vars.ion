echo '# ANCHOR: string_variables'
# The CI can not handle deletions properly.
mkdir -p _tmp _tmp/t1 _tmp/t2
cd _tmp
let filelist = *
echo $filelist
cd ..
echo '# ANCHOR_END: string_variables'
echo '# ANCHOR: string_slicing'
let foo = "Hello, World"
echo $foo[..5]
echo $foo[7..]
echo $foo[2..9]
echo '# ANCHOR_END: string_slicing'
echo '# ANCHOR: string_concatenation'
let string = world
echo $string
let string ++= !
echo $string
let string ::= "Hello, "
echo $string
echo '# ANCHOR_END: string_concatenation'
echo '# ANCHOR: string_operations'
let string1 = Doctor
let string2 = order
echo $string1
echo $string2
let string1 ++= "'s"
let string2 ++= "s"
echo $string1
echo $string2
let string1 ++= " "
let string2 ::= $string1
echo $string2
echo '# ANCHOR_END: string_operations'
