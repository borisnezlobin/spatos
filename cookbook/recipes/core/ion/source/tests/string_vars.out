# ANCHOR: string_variables
t1 t2
# ANCHOR_END: string_variables
# ANCHOR: string_slicing
Hello
World
llo, Wo
# ANCHOR_END: string_slicing
# ANCHOR: string_concatenation
world
world!
Hello, world!
# ANCHOR_END: string_concatenation
# ANCHOR: string_operations
Doctor
order
Doctor's
orders
Doctor's orders
# ANCHOR_END: string_operations
